jiant/tasks/lib/templates/entailment_two_classes.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import torch
+from dataclasses import dataclass
+from typing import List
+
+from jiant.tasks.core import (
+    BaseExample,
+    BaseTokenizedExample,
+    BaseDataRow,
+    BatchMixin,
+    Task,
+    TaskTypes,
+)
+from jiant.tasks.lib.templates.shared import double_sentence_featurize, labels_to_bimap
+from jiant.utils.python.io import read_jsonl
+
+
+@dataclass
+class Example(BaseExample):
+    guid: str
+    input_premise: str
+    input_hypothesis: str
+    label: str
+
+    def tokenize(self, tokenizer):
+        return TokenizedExample(
+            guid=self.guid,
+            input_premise=tokenizer.tokenize(self.input_premise),
+            input_hypothesis=tokenizer.tokenize(self.input_hypothesis),
+            label_id=EntailmentTwoClassesTask.LABEL_TO_ID[self.label],
+        )
+
+
+@dataclass
+class TokenizedExample(BaseTokenizedExample):
+    guid: str
+    input_premise: List
+    input_hypothesis: List
+    label_id: int
+
+    def featurize(self, tokenizer, feat_spec):
+        return double_sentence_featurize(
+            guid=self.guid,
+            input_tokens_a=self.input_premise,
+            input_tokens_b=self.input_hypothesis,
+            label_id=self.label_id,
+            tokenizer=tokenizer,
+            feat_spec=feat_spec,
+            data_row_class=DataRow,
+        )
+
+
+@dataclass
+class DataRow(BaseDataRow):
+    guid: str
+    input_ids: np.ndarray
+    input_mask: np.ndarray
+    segment_ids: np.ndarray
+    label_id: int
+    tokens: list
+
+
+@dataclass
+class Batch(BatchMixin):
+    input_ids: torch.LongTensor
+    input_mask: torch.LongTensor
+    segment_ids: torch.LongTensor
+    label_id: torch.LongTensor
+    tokens: list
+
+
```

Comment:
This file was adapted from the `rte` task.

Reply:
Is there a reason this approach differs from Glue Diagonistics (that inherits from MNLI)?

My reservation is that EntailmentTwoClasses is very specific. If we do want to abstract it out, it could be more general e.g. NLI or even two-input classification.

Reply:
My thoughts were that the GLUE diagnostic specification specifically stated the task was based on MNLI. In this instance, the Ax-g task shares implementation details with two class NLI tasks (specifically RTE). I am open to other abstraction ideas (let me know what you think is best in this case) but I do think it is important to do some variation of inheritance here.

Reply:
Let's follow the format used for GLUE diagnostics and inherit from the RTE task in that case?

Reply:
I'm thinking we can refactor all the classification tasks into a shared implementation (like SQuAD and Multiple Choice) in a later PR.

Reply:
addressed in d5172b9b0ade873782966f105c920c1a7ba2247d